### PR TITLE
fix(OYASUMIVR-20): preserve fractional slider input

### PR DESCRIPTION
**Pull request by Raphiiko**
Typing a fractional chaperone fade distance silently stored a truncated integer. For example, `0.75` became `0`, and `1.4` became `1`.

The shared slider input parsed typed text with `parseInt`, which discards decimals. Every later layer already accepts fractional numbers.

The input now parses decimals, snaps them to the configured step relative to the configured minimum, handles floating-point ties with an epsilon, and clamps the result.

Validated with temporary component checks for chaperone values, integer and decimal steps, halfway and boundary values, non-zero minimums, clamping, empty input, and invalid input. Also passed Prettier, project ESLint, and the development UI build. The repository has no test builder or existing specs, so the component checks were temporary and are not committed.

### CODE REVIEW

File: src-ui/app/components/slider-setting/slider-setting.component.ts [hunk 1]
```diff
@@ -66,8 +66,8 @@ export class SliderSettingComponent implements OnInit, OnChanges {
     this.input$
       .pipe(debounceTime(300), takeUntilDestroyed(this.destroyRef))
       .subscribe((strValue) => {
-        let value = parseInt(strValue, 10);
-        if (isNaN(value)) return;
+        let value = parseFloat(strValue);
```
[the remark this answers is not in the document]
Comment (Raphiiko) [reply src-ui/app/components/slider-setting/slider-setting.component.ts:70]: Fixed in c5740f0f: typed values now use the slider step precision before clamping. Checked step 1 (`2.5` emits `3`), step 0.1 (`0.75` emits `0.8`), and the chaperone step 0.01 cases.

**Comment by Raphiiko** [comment]
@coderabbitai review

**Comment by Raphiiko** [comment]
@codex review

File: src-ui/app/components/slider-setting/slider-setting.component.ts [hunk 2]
```diff
@@ -66,9 +67,13 @@ export class SliderSettingComponent implements OnInit, OnChanges {
     this.input$
       .pipe(debounceTime(300), takeUntilDestroyed(this.destroyRef))
       .subscribe((strValue) => {
-        let value = parseInt(strValue, 10);
-        if (isNaN(value)) return;
-        value = Math.max(this.min, Math.min(this.max, value));
+        let value = parseFloat(strValue);
+        if (!Number.isFinite(value)) return;
+        value = clamp(
+          ensurePrecision(Math.round(value / this.step) * this.step, floatPrecision(this.step)),
```
[the remark this answers is not in the document]
Comment (Raphiiko) [reply src-ui/app/components/slider-setting/slider-setting.component.ts]: Fixed in 5b7118c5: the quotient is rounded to step precision before Math.round. Checked 0.145 at step 0.01 emits 0.15.

File: src-ui/app/components/slider-setting/slider-setting.component.ts [hunk 3]
```diff
@@ -66,9 +67,13 @@ export class SliderSettingComponent implements OnInit, OnChanges {
...
+          this.min,
+          this.max
+        );
```
[the remark this answers is not in the document]
Comment (Raphiiko) [reply src-ui/app/components/slider-setting/slider-setting.component.ts:79]: Fixed in 5b7118c5 with the same precision-safe quotient normalization. Checked 0.15 at step 0.1 emits 0.2.

**Comment by Raphiiko** [comment]
@coderabbitai review

**Comment by Raphiiko** [comment]
@codex review

**Comment by Raphiiko** [comment]
@coderabbitai review

**Comment by Raphiiko** [comment]
@codex review

File: src-ui/app/components/slider-setting/slider-setting.component.ts [hunk 4]
```diff
@@ -66,9 +67,17 @@ export class SliderSettingComponent implements OnInit, OnChanges {
...
       .subscribe((strValue) => {
-        let value = parseInt(strValue, 10);
-        if (isNaN(value)) return;
-        value = Math.max(this.min, Math.min(this.max, value));
+        let value = parseFloat(strValue);
+        if (!Number.isFinite(value)) return;
+        const precision = floatPrecision(this.step);
+        value = clamp(
+          ensurePrecision(
+            Math.round(ensurePrecision(value / this.step, precision)) * this.step,
```
[the remark this answers is not in the document]
Comment (Raphiiko) [reply src-ui/app/components/slider-setting/slider-setting.component.ts]: Fixed in b0490de7: the quotient now uses a magnitude-scaled epsilon without pre-rounding. Checked 0.149 emits 0.1 and 0.15 emits 0.2 at step 0.1.

File: src-ui/app/components/slider-setting/slider-setting.component.ts [hunk 5]
```diff
@@ -66,9 +67,16 @@ export class SliderSettingComponent implements OnInit, OnChanges {
...
-        let value = parseInt(strValue, 10);
-        if (isNaN(value)) return;
-        value = Math.max(this.min, Math.min(this.max, value));
+        let value = parseFloat(strValue);
+        if (!Number.isFinite(value)) return;
+        const precision = floatPrecision(this.step);
+        const quotient = value / this.step;
+        const epsilon = Number.EPSILON * 4 * Math.max(1, Math.abs(quotient));
+        value = clamp(
+          ensurePrecision(Math.round(quotient + epsilon) * this.step, precision),
```
[the remark this answers is not in the document]
Comment (Raphiiko) [reply src-ui/app/components/slider-setting/slider-setting.component.ts]: Fixed in 04c3ba3f: rounding now uses value minus min, adds min back, and normalizes to the precision of both min and step. Checked min 0.25 with step 0.1.

**Comment by Raphiiko** [comment]
@coderabbitai review

**Comment by Raphiiko** [comment]
@codex review
